R/grid_code_paths.R

```diff
@@ -349,16 +475,17 @@ tune_grid_loop_iter <- function(split,
 
 # ------------------------------------------------------------------------------
 
-tune_grid_loop_iter_safely <- function(split,
+tune_grid_loop_iter_safely <- function(fn_tune_grid_loop_iter,
```

Comment:
is there a way to split what this function does into separate functions
- make the safely version of the passed iter function and return it
- gather grid results
- postprocess errors, notes, etc.

Reply:
Okay, so I took another look at this.

I don't think this is actually going to work, mainly because of line 517 which is:

```r
notes <- log_problems(notes, control, split, "internal", problems)
```

We log problems about the iteration results _as they occur_.

If we were to move the post processing step out of the foreach loop, then we'd be forced to log all of the iteration result problems _after_ all of the iterations have run. This would probably result in messages that look out of order since `fn_tune_grid_loop_iter()` itself can also log problems.

i.e. we might get a message on iteration 3 from inside `fn_tune_grid_loop_iter()`, but if iteration 2 fails then the message for that (handled by the safely wrapper) would show up _after_ the message about iteration 3 if we move the post processing out of the foreach loop.

Since we can't cleanly separate the pre/iter/post calls into 3 pieces, I feel like it is probably best to just keep them together as is for now